staging/cse/windows/debug/collect-windows-logs.ps1

```diff
@@ -152,32 +152,77 @@ Get-CimInstance win32_pagefileusage | Format-List * | Out-File -Append "$ENV:TEM
 $netLogs = Get-ChildItem (Get-ChildItem -Path c:\k\debug -Directory | Sort-Object LastWriteTime -Descending | Select-Object -First 1).FullName | Select-Object -ExpandProperty FullName
 $paths += $netLogs
 $paths += "c:\AzureData\CustomDataSetupScript.log"
 
 # log containerd containers (this is done for docker via networking collectlogs.ps1)
-Write-Host "Collecting Containerd running containers"
+Write-Host "Collecting Containerd info from ctr"
+$ctrLogsDirectory = "$ENV:TEMP\$timeStamp-ctr-logs"
 $res = Get-Command ctr.exe -ErrorAction SilentlyContinue
 if ($res) {
-  Write-Host "Collecting Containerd running containers - containers"
-  & ctr.exe -n k8s.io c ls > "$ENV:TEMP\$timeStamp-containerd-containers.txt"
-  $paths += "$ENV:TEMP\$timeStamp-containerd-containers.txt"
+  New-Item -Type Directory $ctrLogsDirectory
+
+  Write-Host "Collecting ctr plugin ls"
+  & ctr.exe -n k8s.io plugin ls > "$ctrLogsDirectory\$timeStamp-containerd-plugin.txt"
+
+  Write-Host "Collecting ctr containers"
+  & ctr.exe -n k8s.io c ls > "$ctrLogsDirectory\$timeStamp-containerd-containers.txt"
+
+  Write-Host "Collecting ctr tasks"
+  & ctr.exe -n k8s.io t ls > "$ctrLogsDirectory\$timeStamp-containerd-tasks.txt"
+
+  Write-Host "Collecting ctr content ls"
+  & ctr.exe -n k8s.io content ls > "$ctrLogsDirectory\$timeStamp-containerd-content.txt"
 
-  Write-Host "Collecting Containerd running containers - tasks"
-  & ctr.exe -n k8s.io t ls > "$ENV:TEMP\$timeStamp-containerd-tasks.txt"
-  $paths += "$ENV:TEMP\$timeStamp-containerd-tasks.txt"
+  Write-Host "Collecting ctr image ls"
+  & ctr.exe -n k8s.io image ls > "$ctrLogsDirectory\$timeStamp-containerd-image.txt"
 
-  Write-Host "Collecting Containerd running containers - snapshot"
-  & ctr.exe -n k8s.io snapshot ls > "$ENV:TEMP\$timeStamp-containerd-snapshot.txt"
-  $paths += "$ENV:TEMP\$timeStamp-containerd-snapshot.txt"
+  Write-Host "Collecting ctr snapshot ls"
+  & ctr.exe -n k8s.io snapshot ls > "$ctrLogsDirectory\$timeStamp-containerd-snapshot.txt"
+
+  Write-Host "Collecting ctr snapshot tree"
+  & ctr.exe -n k8s.io snapshot tree > "$ctrLogsDirectory\$timeStamp-containerd-snapshot-tree.txt"
+
+  Write-Host "Collecting ctr snapshot info for each snapshot"
+  $snapshotsList = & 'C:\Program Files\containerd\ctr.exe' -n k8s.io snapshot ls
+  foreach ($snapshot in $snapshotsList) {
+    $snapshotId = ($snapshot.Split(" ")[0])
+    $fileName = ($snapshotId.Split(":")[1])
+    if ($fileName.length -gt 0) {
+      & 'C:\Program Files\containerd\ctr.exe' -n k8s.io snapshot info $snapshotId > "$ctrLogsDirectory\$timeStamp-containerd-snapshot-info-$fileName.txt"
+    }
+  }
+  $paths += $ctrLogsDirectory
 }
 else {
   Write-Host "ctr.exe command not available"
 }
 
+# collect disk usage information
+if (!(Test-Path "C:\k\debug\DU\du.exe" -PathType Leaf)) {
+  curl -Uri https://download.sysinternals.com/files/DU.zip -UseBasicParsing -OutFile "C:\k\debug\DU.zip"
+  Expand-Archive "C:\k\debug\DU.zip"
+}
+pushd DU
```

Comment:
![image](https://github.com/Azure/AgentBaker/assets/62196586/00ddc552-eb7d-4787-988d-8581388f2eb6)

Cannot run twice. Suggest using absolute path for `du.exe`, e.g.,
```
if (!(Test-Path "c:\k\debug\du.exe")) {
if (Test-Connect "https://download.sysinternals.com/files/DU.zip" succeed) {
  curl -OutFile "$ENV:TEMP\$timeStamp-du.zip"
  Expand-Archive "$ENV:TEMP\$timeStamp-du.zip" "$ENV:TEMP\$timeStamp-du" -Force
  copy "$ENV:TEMP\$timeStamp-du" "c:\k\debug\du.exe"
} else {
  Write-Host "cannot download du"
}
}
// downloads succeed 
if (Test-Path "c:\k\debug\du.exe") {
  c:\k\debug\du.exe /accepteula
  c:\k\debug\du.exe -l 1 C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\
}
```

Reply:
No hurry to change it, I think we should involve @AbelHu and PM to check should we download it or cache in VHD.
If you cannot wait, you can split this PR into two PRs and merge the `cri` changes first.

Reply:
I think that we need to cache the exe. The exe is published and signed by Microsoft so I think that it is good to cache it.

Reply:
OK. I have this PR to cache du.exe - https://github.com/Azure/AgentBaker/pull/3988

After that, the recommend code is:
```
if (Test-Path "c:\tools\du.exe") {
  c:\tools\du.exe /accepteula 
  c:\tools\du.exe -l 1 C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\ 
}
```